b2sdk/bucket.py

```diff
@@ -231,7 +231,9 @@ def get_file_info_by_id(self, file_id: str) -> FileVersion:
         :param str file_id: the id of the file who's info will be retrieved.
         :rtype: generator[b2sdk.v1.FileVersionInfo]
         """
-        return self.FILE_VERSION_FACTORY.from_api_response(self.api.get_file_info(file_id))
+        return self.FILE_VERSION_FACTORY.from_api_response(
```

Comment:
(notice)

in your proposition api is exposing file version factory using bucket.FILE_VERSION_FACTORY and then to factory you need to pass api **instance** - to make this code look **right** I would say you need to write:
```
self.api.file_version_factory().from_api_response(self.api, ...)
```
but as I said - using `self.api` twice looks bad - and my proposition:
```
self.api.file_version_factory.from_api_response(...)
```